Competition2017/src/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -100,10 +100,14 @@ public double calculateHeadingPower() {
         return rotationalPower;
     }
     
+    private double getYDistance() {
+        return poseSubsystem.getRobotOrientedTotalDistanceTraveled().y;
+    }
+    
     @Override
     public boolean isFinished() {
-        double velocity = driveSubsystem.getDistance() - previousPositionInches;
-        previousPositionInches = driveSubsystem.getDistance();
+        double velocity = getYDistance() - previousPositionInches;
```

Comment:
And this should be automatically calculated in the periodic data updates with elapsed time; don't do any measurement in the `isFinished` method.